internal/rbd/group_controllerserver.go

```diff
@@ -130,6 +132,24 @@ func (cs *ControllerServer) CreateVolumeGroupSnapshot(
 			"failed to get existing one with name %q: %v", vgsName, err)
 	}
 
+	creds, err := util.NewUserCredentials(req.GetSecrets())
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	errList := make([]error, 0)
+	for _, volume := range volumes {
+		err = volume.PrepareVolumeForSnapshot(ctx, creds)
+		if err != nil {
+			errList = append(errList, err)
+		}
+	}
+	if len(errList) > 0 {
+		return nil, status.Errorf(
```

Comment:
PrepareVolumeForSnapshot already returning the GRPC error message(s) we are returning Abort error message do why returning abort for all the cases?

Reply:
> PrepareVolumeForSnapshot already returning the GRPC error message(s) we are returning Abort error message do why returning abort for all the cases?

It's now returning the grpc code from first error in the list.
Added a TODO to filter out a higher priority error code.